src/forwardBeamPipeBrazil.cpp

```diff
@@ -0,0 +1,547 @@
+#include <XML/Helper.h>
+
+using namespace std;
+using namespace dd4hep;
+
+static Ref_t create_detector(Detector& det, xml_h e, SensitiveDetector /* sens */)
+{
+
+    using namespace ROOT::Math;
+    xml_det_t x_det    = e;
+    string    det_name = x_det.nameStr();
+    DetElement sdet(det_name, x_det.id());
+    Assembly   assembly(det_name + "_assembly");
+    Material m_Al     = det.material("Aluminum");
+    Material m_Be     = det.material("Beryllium");
+    Material m_SS     = det.material("StainlessSteel");
+    Material m_vac    = det.material("Vacuum");
+    string   vis_name = x_det.visStr();
+
+    PlacedVolume pv_assembly;
+
+    double b0_hadron_tube_inner_r = 2.9 * dd4hep::cm;
+    double b0_hadron_tube_outer_r = 3.1 * dd4hep::cm;
+    double b0_hadron_tube_length  = 120.0 * dd4hep::cm;
+
+    double pipeThickness     = 5.0 * dd4hep::mm;
+
+    struct beampipe_dimensions_t {
+        Double_t length;
+        Double_t innerXRadius;
+        Double_t innerYRadius;
+        Double_t outerXRadius;
+        Double_t outerYRadius;
+        Double_t xCenter;
+        Double_t yCenter;
+        Double_t zCenter;
+        Double_t rotationAngle;
+    };
+
+    std::vector<beampipe_dimensions_t> beampipe_dimensions;
+
+    double globRotationAngle = -0.0454486856; //This is the angle of the proton orbit from the end of B1APF to the beginning of B2PF
+    double crossingAngle     = -0.025; //relevant for the neutral cone
+
+    double b1APFEndPoint_z = 22062.3828 * dd4hep::mm; //location of proton orbit at b1APF exit -- in mm
+    double b1APFEndPoint_x = 654.3372 * dd4hep::mm; //location of proton orbit at b1APF exit -- in mm
+
+    double tmp_endpoint_z = 0.0;
+    double tmp_endpoint_x = 0.0;
+
+    //forumula -> Z = b1APFEndPoint_z+((0.5*elementLengt)*Cos(globRotationAngle))
+    //forumula -> X = b1APFEndPoint_z+((0.5*elementLengt)*Sin(globRotationAngle))
+
+        //------------------------------------------------------------------------------------
+        //Geometry extracted from version 0 VPC drawings shown at the FF preliminary
+        //design review in February 2024 -- CAD model not available as of April 1st, 2024
+        //------------------------------------------------------------------------------------
+
+        //------------------------------------------------------------------------------------
+    //primary pipe after B1APF, before neutral exit window + transition to smaller pipe
+    //rectangular cross-section!!!!!
+    //------------------------------------------------------------------------------------
+
+
+    beampipe_dimensions.push_back({
+
+        .length            = 7615.486 * dd4hep::mm, //from VPC drawings, in mm
+        .innerXRadius      = 275.0 * dd4hep::mm,
+        .innerYRadius      = 175.0 * dd4hep::mm,
+        .outerXRadius      = 0.0,
+        .outerYRadius      = 0.0,
+        .xCenter           = 0.0,
+        .yCenter           = 0.0,
+        .zCenter           = 0.0,
```

Comment:
If you don't set these values (or set them below), you can just leave these lines out. The reason for the `.fieldname` syntax is to allow you to do just that. (Applies file-wide.)

Reply:
This is only true if you initialize them when you define the struct, which I had not done, so removing those lines causes a compiler error. I went ahead and initialized it when I define the struct to remove those lines.